### PR TITLE
JSTL-AdminSearchPage

### PR DESCRIPTION
**Pull request by taniach**

### CODE REVIEW

**Comment by taniach** [comment]
@bamboo3250 please review

File: src/main/java/teammates/common/datatransfer/FeedbackSessionState.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package teammates.common.datatransfer;
+
+public enum FeedbackSessionState {
+    OPEN, CLOSED, PUBLISHED;
+}
```
Comment (damithc) [inline src/main/java/teammates/common/datatransfer/FeedbackSessionState.java]: This is not defined anywhere else? Anyway, the AWAITING state is missing?

**Comment by damithc** [comment]
How come no jsp code is deleted to make way for new jstl code?

**Comment by taniach** [comment]
Sorry I had commented it and forgot to remove. Just removed it.

File: src/main/java/teammates/common/datatransfer/FeedbackSessionState.java (hunk 1 above)
Comment (taniach) [inline src/main/java/teammates/common/datatransfer/FeedbackSessionState.java]: This enum is not defined yet. We use strings from Const.java. 
There is a FeedbackSessionType enum which has STANDARD and PRIVATE.

Comment (damithc) [inline src/main/java/teammates/common/datatransfer/FeedbackSessionState.java]: It is best to define it in one place rather than in multiple locations.

Comment (taniach) [inline src/main/java/teammates/common/datatransfer/FeedbackSessionState.java]: We can create an issue to convert strings to enums in other places.

File: src/main/webapp/jsp/adminSearch.jsp [hunk 2]
```diff
@@ -1,555 +1,24 @@
...
-
-            </div>
-
-
-
-    <jsp:include page="<%=Const.ViewURIs.FOOTER%>" />
-</body>
-</html>
\ No newline at end of file
+    <script type="text/javascript" src="/bootstrap/js/bootstrap.min.js"></script>
```
Comment (bamboo3250) [inline src/main/webapp/jsp/adminSearch.jsp]: I think we have `/bootstrap/js/bootstrap.min.js` in `page.tag`

Comment (taniach) [inline src/main/webapp/jsp/adminSearch.jsp]: @bamboo3250 apparently this is getting overwritten so if I don't include it, bootstrap doesn't work for the tooltip

File: src/main/java/teammates/ui/controller/AdminSearchPageData.java [hunk 3]
```diff
@@ -38,7 +50,185 @@
...
+        String courseId = instructor.courseId;
+        String googleId = instructor.googleId;
+        String googleIdLink = instructorHomaPageLinkMap.get(instructor.googleId);
+        String institute = instructorInstituteMap.get(instructor.getIdentificationString());
+        String viewRecentActionsId = createViewRecentActionsId(instructor);
+        String email = instructor.email;
+        String courseJoinLink = instructorCourseJoinLinkMap.get(instructor.getIdentificationString());
+        
+        return new AdminSearchInstructorRow(id, name, courseName, courseId, googleId, googleIdLink, 
+                                                institute, viewRecentActionsId, email, courseJoinLink);
```
Comment (bamboo3250) [inline src/main/java/teammates/ui/controller/AdminSearchPageData.java]: indent the `institute` to have the same column as `id`

File: src/main/java/teammates/ui/controller/AdminSearchPageData.java [hunk 4]
```diff
@@ -38,7 +50,185 @@
...
+        
+        List<AdminSearchStudentFeedbackSession> openFeedbackSessions = 
+                                        createFeedbackSessionsList(student, FeedbackSessionState.OPEN);
+        List<AdminSearchStudentFeedbackSession> closedFeedbackSessions = 
+                                        createFeedbackSessionsList(student, FeedbackSessionState.CLOSED);
+        List<AdminSearchStudentFeedbackSession> publishedFeedbackSessions = 
+                                        createFeedbackSessionsList(student, FeedbackSessionState.PUBLISHED);
+        
+        return new AdminSearchStudentRow(id, name, institute, courseName, courseId, section, 
+                                        team, googleId, email, comments, viewRecentActionsId,
```
Comment (bamboo3250) [inline src/main/java/teammates/ui/controller/AdminSearchPageData.java]: it should be at the same column as `id`?

File: src/main/java/teammates/ui/template/AdminSearchStudentRow.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    private String viewRecentActionsId;
+    
+    private AdminSearchStudentLinks links;
+    
+    private List<AdminSearchStudentFeedbackSession> openFeedbackSessions;
+    private List<AdminSearchStudentFeedbackSession> closedFeedbackSessions;
+    private List<AdminSearchStudentFeedbackSession> publishedFeedbackSessions;
+    
+    public AdminSearchStudentRow(String id, String name, String institute, String courseName,
+                                    String courseId, String section, String team, String googleId,
```
Comment (bamboo3250) [inline src/main/java/teammates/ui/template/AdminSearchStudentRow.java]: the indentation is little off here

**Comment by taniach** [comment]
@bamboo3250 ready for review

**Pull request merged**